### PR TITLE
add ros deploy support

### PR DESCRIPTION
**Pull request by tanhe123**

### CODE REVIEW

File: bin/fun-deploy.js [hunk 1]
```diff
@@ -22,6 +22,8 @@ program
 
   .option('-t, --template [template]', 'path of fun template file.')
   .option('-c, --only-config', 'Update only configuration flags')
+  .option('--use-ros', 'deploy resources using ros')
```
Comment (vangie) [inline bin/fun-deploy.js]: deploy resources using ROS

File: bin/fun-deploy.js [hunk 2]
```diff
@@ -22,6 +22,8 @@ program
...
+  .option('--stack-name <stackName>', 'the name of the ros stack')
```
Comment (vangie) [inline bin/fun-deploy.js]: ros => ROS

File: lib/deploy/deploy-by-tpl.js [hunk 3]
```diff
@@ -618,7 +603,48 @@ async function deploy(tplPath, context) {
...
+
+  const baseDir = path.resolve(tplPath, '..');
+
+  const dirName = path.basename(baseDir);
+
+  const DEFAULT_STACK_NAME = `fun_default_stack_for_${dirName}`;
+
+  if (context.useRos) {
+    if (context.stackName) {
+      console.log(yellow(`deploy using ros, stack name is ${context.stackName}`));
```
Comment (vangie) [inline lib/deploy/deploy-by-tpl.js]: deploying by ROS, The Stack name is

File: lib/deploy/deploy-support-ros.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+async function updateChangeSet(rosClient, region, stackName, stackId, tpl) {
+  try {
+    const params = {
+      'RegionId': region,
+      'ChangeSetName': generateChangeSetName(),
+      'StackId': stackId,
+      'ChangeSetType': 'UPDATE',
+      'Description': 'generate by fun',
```
Comment (vangie) [inline lib/deploy/deploy-support-ros.js]: generated by Funcraft

File: lib/deploy/deploy-support-ros.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  }
+}
+
+async function createChangeSet(rosClient, region, stackName, tpl) {
+  const params = {
+    'RegionId': region,
+    'ChangeSetName': generateChangeSetName(),
+    'StackName': stackName,
+    'ChangeSetType': 'CREATE',
+    'Description': 'generate by fun',
```
Comment (vangie) [inline lib/deploy/deploy-support-ros.js]: generated by Funcraft

File: bin/fun-deploy.js (hunk 1 above)
Comment (tanhe123) [reply bin/fun-deploy.js]: done

File: bin/fun-deploy.js (hunk 2 above)
Comment (tanhe123) [reply bin/fun-deploy.js]: done

File: lib/deploy/deploy-by-tpl.js (hunk 3 above)
Comment (tanhe123) [reply lib/deploy/deploy-by-tpl.js]: done

File: lib/deploy/deploy-support-ros.js (hunk 4 above)
Comment (tanhe123) [reply lib/deploy/deploy-support-ros.js]: done

File: lib/deploy/deploy-support-ros.js (hunk 5 above)
Comment (tanhe123) [reply lib/deploy/deploy-support-ros.js]: done

File: bin/fun-deploy.js [hunk 6]
```diff
@@ -22,6 +22,8 @@ program
 
   .option('-t, --template [template]', 'path of fun template file.')
   .option('-c, --only-config', 'Update only configuration flags')
+  .option('--use-ros', 'deploy resources using ROS')
```
Comment (vangie) [inline bin/fun-deploy.js]: deploy =》 Deploy
感觉上下文有多处首字母没有大写的情况

File: bin/fun-deploy.js [hunk 7]
```diff
@@ -22,6 +22,8 @@ program
 
   .option('-t, --template [template]', 'path of fun template file.')
```
Comment (vangie) [inline bin/fun-deploy.js]: The path of

Comment (tanhe123) [reply bin/fun-deploy.js]: done

File: bin/fun-deploy.js (hunk 6 above)
Comment (tanhe123) [reply bin/fun-deploy.js]: done

File: bin/fun-init.js [hunk 8]
```diff
@@ -48,11 +48,11 @@ program
   .name('fun init')
   .usage('[options] [template]')
   .description('Initialize a new project based on a template. A template can be a folder containing template metadata and boilerplate files, a name of a pre-built template, or a url that resolves to a template. You can find more information about template at https://yq.aliyun.com/articles/674364.')
-  .option('-o, --output-dir [path]', 'where to output the initialized app into', '.')
-  .option('-n, --name [name]', 'name of your project to be generated as a folder', '')
-  .option('-m, --merge [merge]', 'merge into the template.[yml|yaml] file if it already exist', false)
-  .option('--no-input', 'disable prompting and accept default values defined template config')
-  .option('-V, --var [vars]', 'template variable', parseVars)
+  .option('-o, --output-dir [path]', 'Where to output the initialized app into', '.')
+  .option('-n, --name [name]', 'Name of your project to be generated as a folder', '')
```
Comment (vangie) [inline bin/fun-init.js]: The name of

File: bin/fun-deploy.js (hunk 6 above)
Comment (tanhe123) [reply bin/fun-deploy.js]: 全部统一大写了

File: bin/fun-init.js (hunk 8 above)
Comment (tanhe123) [reply bin/fun-init.js]: done

**Pull request merged**